### PR TITLE
Added a starboard clone as an example

### PR DESCRIPTION
**Pull request by Piecuuu**
## Notice
- [x] I *understand* the code that I have edited, and have the means
to test it before making changes to Concord.

## What?

I added a starboard clone as an example.

## Why?

By adding a more complex example, users can learn about the more complicated side of Concord with an example. 

## How?

Made an example.

## Testing?

It works on many machines. I've not yet found any risks, untested conditions or edge cases. 

## Anything Else?

We could add a guide for installing dependencies on OpenBSD or FreeBSD.

### CODE REVIEW

**Comment by ProfessorHugeD** [comment]
This project looks like it's dead and no longer maintained.

**Comment by HackerSmacker** [comment]
Works on:
- FreeBSD 13
- OpenBSD 7.1
- Windows 10 (Cygwin et al)
- GNU/Linux circa Nov 2023
- OS X 13.1
No issues with Postgres. This bot is now running in production!

**Comment by Piecuuu** [comment]
Great to hear that!

File: examples/starboard/src/bot.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  // I think we need this, but I'm not sure.
+  discord_add_intents(client, DISCORD_EV_MESSAGE_REACTION_ADD);
+
+  // This is here because we have to get the url from the config right after we initialise the client 
+  star_db_connect(client);
+
+  discord_set_on_ready(client, &star_bot_on_ready);
+  discord_set_on_message_reaction_add(client, &star_main_handler);
+  discord_set_on_interaction_create(client, &star_commands_handler);
+  discord_run(client);
```
Comment (ThePedroo) [inline examples/starboard/src/bot.c:35]: You are not cleaning up the client resources

File: examples/starboard/src/bot.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#include <string.h>
+#include "db.h"
+#include "commands.h"
+#include "starboard.h"
+
+static void star_bot_on_ready(struct discord* client, const struct discord_ready* event) {
+  char* discriminator = event->user->discriminator;
+  char* username = event->user->username;
+
+  if(strncmp(discriminator, "0", strlen(discriminator)) == 0) { // This is for the time, when Discord bots will lose their discriminators
```
Comment (ThePedroo) [inline examples/starboard/src/bot.c:13]: Perhaps remove and just add when they really do -- If they do?

Comment (ThePedroo) [reply examples/starboard/src/bot.c:13]: Also the syntax used in Concord has a space between `if` and `(`, might want to follow it here too. (as well as other things)

**Review by ThePedroo — COMMENTED** [review]
LGTM, just some few things. Overall you may want to make the comments more objective, but for me, as long as people can use it as example to build more advanced bots, it's awesome! Good job

**Comment by ThePedroo** [comment]
Oh, and you may want to run it against valgrind

File: examples/starboard/src/bot.c (hunk 2 above)
Comment (Piecuuu) [reply examples/starboard/src/bot.c:13]: Discord definetly will phase out tags for bots, but even if they don't, I think It's good future-proofing and can be easly modified once the transition happens.

**Comment by lcsmuller** [comment]
This is too complex to be added as an example. But feel free to open a PR adding a section in README with community-driven bots, and add a link to your bot

**Pull request closed without merging**